libraries/AP_Scripting/examples/MinFixType.lua

```diff
@@ -0,0 +1,187 @@
+Checks for mission running and commands a hold/pause if the GPS fix type is less
+than a threshold value.
+
+CAUTION: This script is capable of engaging and disengaging autonomous control
+of a vehicle.  Use this script AT YOUR OWN RISK.
+
+-- Yuri -- Aug 2021, revised Apr 2022
+
+LICENSE - GNU GPLv3 https://www.gnu.org/licenses/gpl-3.0.en.html
+------------------------------------------------------------------------------]]
+
+local SCRIPT_NAME = 'MinFixType'
+
+--------  MAVLINK/AUTOPILOT 'CONSTANTS'  --------
+local ROVER_MODE_MANUAL       =  0
+local ROVER_MODE_HOLD         =  4
+local ROVER_MODE_AUTO         = 10
+local MAV_SEVERITY_WARNING    =  4
+local MAV_SEVERITY_INFO       =  6
+
+--------  USER EDITABLE GLOBALS  --------
+local GPS_INSTANCE = 0                 -- GPS to monitor (moving base, most likely)
+local MIN_FIX_TYPE = 6                 -- see table below
+local MSN_PAUSE_MODE     = ROVER_MODE_MANUAL  -- mode to command when GPS fix is inadequate
+local THR_SAFEGUARD_MODE = ROVER_MODE_HOLD    -- mode to command if mission paused with non-zero throttle
+local BAD_FIX_TIMEOUT  = 1600          -- how long a bad fix type must be present before pausing the mission
+local GOOD_FIX_TIMEOUT =  600          -- how long a good fix type must be present before resuming the mission
+local RUN_INTERVAL_MS  =  200          -- (ms) how often to run this script (50-250 should work fine)
+local VERBOSE_MODE     =    2  -- 0 to suppress all GCS messages,
+                               -- 1 for normal status messages
+                               -- 2 for additional debug messages
+local MSG_NORMAL = 1
+local MSG_DEBUG  = 2
+
+local FIX_TYPES = {
+    [0] = 'No GPS',  -- Lua arrays are 1 based unless you specify discrete indices like this
+    [1] = 'No Fix',
+    [2] = '2D Fix',
+    [3] = '3D Fix',
+    [4] = 'DGPS Fix',
+    [5] = 'RTK Float',
+    [6] = 'RTK Fixed',
+    [7] = 'Static Fixed',
+    [8] = 'PPP, 3D'}
+
+local MODE_THRESHOLDS = {1231, 1361, 1491, 1621, 1750, 2050}
+
+local MODE_CH  = param:get('MODE_CH')
+local THR_CH   = param:get('RCMAP_THROTTLE')
+local THR_TRIM = param:get(string.format('RC%d_TRIM', THR_CH))
+local THR_DZ   = param:get(string.format('RC%d_DZ', THR_CH))
+
+-- wrapper for gcs:send_text()
+local function gcs_msg(msg_type, severity, txt)
+    if type(msg_type) == 'string' then
+    -- allow just a string to be passed for simple/routine messages
+        txt      = msg_type
+        msg_type = MSG_NORMAL
+        severity = MAV_SEVERITY_INFO
+    end
+    if msg_type <= VERBOSE_MODE then
+        gcs:send_text(severity, string.format('%s: %s', SCRIPT_NAME, txt))
+    end
+end
+
+-- return RC transmitter selected mode
+local function get_user_mode()
+    local pwm = rc:get_pwm(MODE_CH)
+    local mode_num = 6
+    for i, threshold in pairs(MODE_THRESHOLDS) do
+        if (pwm < threshold) then
+            mode_num = i
+            break
+        end
+    end
+    return tonumber(param:get('MODE' .. mode_num))

```

Comment:
Ideally we would cache these params, either there values or param handles. Just saves doing the lookup by string at 5hz. 

Did you find you needed the `tonumber`? It should always return a number (so long as it finds the param)

Reply:
Agree that's a better method for this particular script where those params are unlikely to change during runtime.

I think `tonumber` was a workaround at some point but probably overcome by events and unnecessary at this point.

Can modify to suit.